core/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -654,6 +654,15 @@ BlobContainer blobContainer() {
 
     protected void writeIndexGen(final RepositoryData repositoryData) throws IOException {
         assert isReadOnly() == false; // can not write to a read only repository
+        final long currentGen = latestIndexBlobId();
+        if (currentGen != repositoryData.getGenId()) {
+            // the index file was updated by a concurrent operation, so we were operating on stale
+            // repository data
+            throw new RepositoryException(metadata.name(), "concurrent modification of the index-N file, expected current generation [" +
```

Comment:
That seems a bit drastic. On the other hand if we are here, it might mean that while we were creating this snapshot somebody might have deleted some snapshots and therefore the segments that we think are there might be no longer there. So, at this point we cannot assume that this snapshot is complete. Right?